src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala

```diff
@@ -24,31 +29,36 @@ trait LdapSupport {
     * @tparam T
     * @return
     */
-  protected def ldapSearchStream[T](baseDn: String, searchScope: SearchScope, filters: Filter*)(unmarshaller: Entry => T): Stream[T] =
-    filters.flatMap { filter =>
-      val search = new SearchRequest(baseDn, searchScope, filter)
-      val entrySource = new LDAPEntrySource(ldapConnectionPool.getConnection, search, true)
-      ldapEntrySourceStream(entrySource)(unmarshaller)
-    }.toStream
-
-  // this is the magic recursive stream generator
-  private def ldapEntrySourceStream[T](entrySource: LDAPEntrySource)(unmarshaller: Entry => T): Stream[T] =
-    Try(Option(entrySource.nextEntry)) match {
-      case Success(None) =>
-        // reached the last element, Stream.empty terminates the stream
-        Stream.empty
-
-      case Success(Some(next)) =>
-        // next element exists, return a Stream starting with unmarshalled next followed by the rest of the stream
-        // (streams are smart and lazily evaluate the second parameter)
-        Stream.cons(unmarshaller(next), ldapEntrySourceStream(entrySource)(unmarshaller))
-
-      case Failure(ldape: EntrySourceException)
-          if ldape.getCause.isInstanceOf[LDAPException] &&
-            ldape.getCause.asInstanceOf[LDAPException].getResultCode == ResultCode.NO_SUCH_OBJECT =>
-        Stream.empty // the base dn does not exist, treat as empty search
-      case Failure(regrets) => throw regrets
+  protected def ldapSearchStream[T](baseDn: String, searchScope: SearchScope, filters: Filter*)(unmarshaller: Entry => IO[T]): fs2.Stream[IO, T] = {
+    for {
+      filter <- fs2.Stream(filters:_*)
```

Comment:
I think this way, you're apply filter one at a time, and doing a search based on each individual filter. Looks to me you should pass the whole filter list to `ldapSearch` directly